api/server/sdk/cluster_pair.go

```diff
@@ -0,0 +1,135 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/cluster"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ClusterPairServer is an implementation of the gRPC OpenStorageClusterServer interface
+type ClusterPairServer struct {
+	server *Server
+}
+
+func (s *ClusterPairServer) cluster() cluster.Cluster {
+	return s.server.cluster()
+}
+
+// Create a new cluster with remote pair
+func (s *ClusterPairServer) Create(
+	ctx context.Context,
+	req *api.SdkClusterPairCreateRequest,
+) (*api.SdkClusterPairCreateResponse, error) {
+
+	if req.GetRequest() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Must supply valid request")
+	}
+
+	resp, err := s.cluster().CreatePair(req.GetRequest())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot create cluster with remote pair %s : %v",
+			req.GetRequest().GetRemoteClusterIp(), err)
+	}
+
+	return &api.SdkClusterPairCreateResponse{
+		Result: resp,
+	}, nil
+}
+
+// Inspect information about a cluster pair
+func (s *ClusterPairServer) Inspect(
+	ctx context.Context,
+	req *api.SdkClusterPairInspectRequest,
+) (*api.SdkClusterPairInspectResponse, error) {
+
+	if len(req.GetId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply cluster ID")
+	}
+	resp, err := s.cluster().GetPair(req.GetId())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot Get cluster information for %s : %v", req.GetId(), err)
+	}
+	return &api.SdkClusterPairInspectResponse{
+		Result: resp,
+	}, nil
+}
+
+// Enumerate returns list of cluster pairs
+func (s *ClusterPairServer) Enumerate(
+	ctx context.Context,
+	req *api.SdkClusterPairEnumerateRequest,
+) (*api.SdkClusterPairEnumerateResponse, error) {
+
+	resp, err := s.cluster().EnumeratePairs()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot list cluster pairs : %v", err)
+	}
+	return &api.SdkClusterPairEnumerateResponse{
+		Result: resp,
+	}, nil
+}
+
+// Token gets the authentication token for this cluster
+func (s *ClusterPairServer) Token(
+	ctx context.Context,
+	req *api.SdkClusterPairTokenRequest,
+) (*api.SdkClusterPairTokenResponse, error) {
+
+	resp, err := s.cluster().GetPairToken(false)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot generate token : %v", err)
+	}
+	return &api.SdkClusterPairTokenResponse{
+		Result: resp,
+	}, nil
+}
+
+// ClearToken gets the authentication token for this cluster
+func (s *ClusterPairServer) ClearToken(
```

Comment:
This API doesn't make sense. If it's to Clear the token it shouldn't return the token.

Reply:
I agree that it should not return a token. Is that what happens when GetToken in golang is set with the boolean to clear it? if it does, then yeah, we should not return a token.

Reply:
There is no ClearToken() in the golang. While getting a token you can specify if you want to reset it before returning. So I'm not sure why this was split up in to 2 APIs here.

Reply:
It does not matter that there is no ClearToken() in golang. The SDK's requirement is to provide a clear message and API to the callers.

Just like we created volume.Clone() (does not exist in golang) and EnumerateWithFilters. Etc.

Reply:
Having a GetToken() with a parameter whether to reset is clear enough.